src/lib/utils/breadcrumbUtils.tsx

```diff
@@ -0,0 +1,69 @@
+import { BreadcrumbItem } from '@/types/dataTypes/drupal/field_type'
+
+export function deriveLastBreadcrumbFromPath(
+  breadcrumbs: BreadcrumbItem[],
+  breadcrumbTitle: string,
+  currentPath: string,
+  replaceLastItem: boolean
+): BreadcrumbItem[] {
+  const last: BreadcrumbItem = {
+    uri: currentPath,
+    title: breadcrumbTitle,
+    options: [],
+  }
+
+  if (replaceLastItem) {
+    breadcrumbs.splice(breadcrumbs.length - 1, 1, last)
+  } else {
+    breadcrumbs.push(last)
+  }
+
+  return breadcrumbs
+}
+
+// Intended for handling breadcrumbs for learning center content types
+export function deriveLcBreadcrumbs(
+  breadcrumbs: BreadcrumbItem[],
+  breadcrumbTitle: string,
+  currentPath: string,
+  lcBreadcrumbsTitleInclude?: boolean
+): BreadcrumbItem[] {
+  const filteredCrumbs: BreadcrumbItem[] = breadcrumbs.filter(
+    (crumb) => crumb.uri !== '/resources'
+  )
+
+  filteredCrumbs.push({
+    uri: currentPath,
+    title: 'Resources and support',
+    options: [],
+  })
+
+  if (lcBreadcrumbsTitleInclude) {
+    filteredCrumbs.push({
+      uri: currentPath,
+      title: breadcrumbTitle,
+      options: [],
+    })
+  }
+
+  return filteredCrumbs
+}
+
+export function transformBreadcrumbs(breadcrumbs) {
+  const transformedCrumbs = breadcrumbs.map((crumb) => {
+    const pathname = new URL(crumb.uri, 'http://va.gov').pathname
```

Comment:
**CONSIDER**
I wonder if this might be useful as an abstraction of its own. `getRelativePathFromUrl`. We'll likely need this again elsewhere, and, while the code is relatively short as it is, it is a bit esoteric (need to know parameters as well as property to grab). It would maybe be useful to have `@/lib/utils/url` with functions like this.

Reply:
That makes sense, broke it out to its own function